lib/mongodb.js

```diff
@@ -615,12 +615,12 @@ MongoDB.prototype.destroyAll = function destroyAll(model, where, callback) {
     where = undefined;
   }
   where = self.buildWhere(model, where);
-  this.collection(model).remove(where || {}, function (err, result) {
-    if (self.debug) {
-      debug('destroyAll.callback', model, where, err, result);
-    }
-    var count = result && result.result && result.result.n || 0;
-    callback && callback(err, count);
+  this.collection(model).remove(where || {}, function(err, results) {
+    if (self.debug)
+      debug('destroyAll.callback', model, where, err, results);
+
+    var affectedCount = (results.result && results.result.n) || 0;
+    callback && callback(err, {count: affectedCount});
```

Comment:
@raymondfeng I'm looking back at this now and are we sure this is correct? I think the bracket causes the affectedCount to be true/false or 0. I think we want `result.result && results.result.n ? results.result.n : 0`.